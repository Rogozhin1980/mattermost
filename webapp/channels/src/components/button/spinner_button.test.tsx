// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {mount, shallow} from 'enzyme';
import type {ComponentProps} from 'react';
import React from 'react';

import SpinnerButton from './spinner_button';

function getBaseProps(): ComponentProps<typeof SpinnerButton> {
    return {
        emphasis: 'primary',
        idleText: 'Idle',
        spinningText: 'Spinning',
    };
}

describe('components/SpinnerButton', () => {
    test('should match snapshot with required props', () => {
        const wrapper = shallow(
            <SpinnerButton
                {...getBaseProps()}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with spinning', () => {
        const wrapper = shallow(
            <SpinnerButton
                {...getBaseProps()}
                spinning={true}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should handle onClick', () => {
        const onClick = jest.fn();

        const wrapper = mount(
            <SpinnerButton
                {...getBaseProps()}
                onClick={onClick}
            />,
        );

        wrapper.find('button').simulate('click');
        expect(onClick).toHaveBeenCalledTimes(1);
    });

    test('should add properties to underlying button', () => {
        const wrapper = mount(
            <SpinnerButton
                {...getBaseProps()}
                testId='my-button-id'
                emphasis='tertiary'
            />,
        );

        const button = wrapper.find('button');

        expect(button).not.toBeUndefined();
        expect(button.type()).toEqual('button');
        expect((button.props() as any)['data-testid']).toEqual('my-button-id');
        expect(button.hasClass('btn')).toBeTruthy();
        expect(button.hasClass('btn-tertiary')).toBeTruthy();
    });
});
