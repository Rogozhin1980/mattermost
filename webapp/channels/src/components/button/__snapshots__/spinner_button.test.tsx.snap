// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SpinnerButton should match snapshot with required props 1`] = `
<Button
  disabled={false}
  emphasis="primary"
  label="Idle"
  loading={false}
/>
`;

exports[`components/SpinnerButton should match snapshot with spinning 1`] = `
<Button
  disabled={true}
  emphasis="primary"
  label="Spinning"
  loading={true}
/>
`;
